### DISCUSSION TITLE
Minimal fybrik installation without all

### DISCUSSION THREAD

**Discussion by lappanna** [fybrik/fybrik#811] [Ideas]
It will be good if fybrik can be installed minimally by default without all components that are not required. Or there can be an option to install minimal fybrik.

**Comment by roee88** [comment]
Please list components that you consider as not required for a minimal install

**Comment by simanadler** [comment]
@lappanna @ronenkat @roee88 Reviewing the main fybrik repository I would suggest that the following components be moved out of the main fybrik repository (and thus would not automatically be installed with fybrik):

1) movement operator 
* module definitions - 
    https://github.com/fybrik/fybrik/tree/master/modules/fybrik-implicit-copy-batch
    https://github.com/fybrik/fybrik/tree/master/modules/fybrik-implicit-copy-stream
* motion controller - https://github.com/fybrik/fybrik/tree/master/manager/controllers/motion

2) all the open source connectors - https://github.com/fybrik/fybrik/tree/master/connectors

3) samples - https://github.com/fybrik/fybrik/tree/master/samples

4) fybrik/third_party/datashim - only needed when doing implicit copies.  However, removing it will require changes to the manager.

@roee88 @ronenkat Removing these would mean that we would have to change the quick start, samples, and CI tests to pull from multiple repositories under the fybrik organization.  Would it make sense to have the following repositories under the fybrik organization?
* connectors
* modules - which describe services in data plane services or services provided by other organizations
* data plane services
* samples

If so, the current fybrik repository becomes the fybrik control plane.

**Reply by shlomitk1** [reply]
@simanadler 
- connectors: deployment is configurable based on configurations in `fybrik `chart values.
- motion controller: there is an argument to the manager that controls the setup.
- datashim is not a part of the `fybrik `chart installation.

**Reply by roee88** [reply]
I think that this list goes beyond the original request but I might be wrong. 

The movement operator is always installed, even if not running the CRDs  and code are there. It would be good to try separating it.

Storage provisioning related CRDs and code might eventually be generalized as modules instead of being dependent on datashim. I suspect that this is too much work for now and would leave it as is for now.

I don't see a problem with samples and connectors being in the repo to fulfill the original request. Cleanup there is required but it's not necessarily related to this discussion.

**Comment by simanadler** [comment]
@lappanna The comments by @shlomitk1 raise a good point.  Are we trying to minimize what is installed, and/or to minimize what code resides in the fybrik control plane repository?  I was assuming both, but perhaps I was wrong.  

One of the advantages of separating the control plane code from the other optional components is that it decreases the footprint when doing security scans, etc.  @roee88 Your thoughts?

**Reply by ronenkat** [reply]
@simanadler moving code out of the Fybrik repo also mean more difficulty in keeping things consistent, impact CI effectiveness, etc.
I think I am ok with (1).
For (2) we can move the Egeria connector, see #845, but not the rest of them.
Lets keep (3) and (4) in the repo to minimize change.

**Reply by simanadler** [reply]
@ronenkat Regarding connectors I think we need to be consistent.  I think we can take two possible approaches:

1) fybrik control plane has a single default connector for each of the connector types, which will be in the fybrik repository.  These are open source and guaranteed to be up to date and work with the current version of the fybrik control plane.  Currently these would be katalog for the Data Catalog and OPA for the policy manager. ?? for the credential manager. All other connectors are stored outside of the main fybrik repository.

Different connectors can be deployed when desired. _Need to include docs and tools for doing this._

2) All connectors reside outside of fybrik and prior to installing fybrik the chosen connectors should be installed.

I suggest we go with #1.  Is everyone in agreement?  @lappanna @roee88 @ronenkat @shlomitk1 

@lappanna You are OK with Ronen's suggestion to leave datashim and samples in the fybrik repository?

**Reply by ronenkat** [reply]
@simanadler agree with your suggestion to go with option 1. I think that after #845 we will reach this option.
For credential we have Vault as the connector.

**Reply by shlomitk1** [reply]
@simanadler Agree with option 1.
